src/teachy_tv.c

```diff
@@ -0,0 +1,1775 @@
+        b _0815B37A\n\
+        .align 2, 0\n\
+    _0815B31C: .4byte gTasks+0x8\n\
+    _0815B320: .4byte gPaletteFade\n\
+    _0815B324: .4byte gTeachyTV_StaticResources+6\n\
+    _0815B328: .4byte gMain\n\
+    _0815B32C: .4byte UseFameCheckerFromMenu\n\
+    _0815B330:\n\
+        movs r0, 0x5\n\
+        bl PlaySE\n\
+        adds r0, r7, 0\n\
+        bl TeachyTvQuitBeginFade\n\
+        b _0815B37A\n\
+    _0815B33E:\n\
+        movs r0, 0x5\n\
+        bl PlaySE\n\
+        ldr r2, _0815B380 @ =gTeachyTV_StaticResources\n\
+        movs r4, 0\n\
+        strb r5, [r2, 0x5]\n\
+        ldrb r0, [r6]\n\
+        adds r1, r2, 0x6\n\
+        adds r2, 0x8\n\
+        bl DestroyListMenu\n\
+        bl TeachyTvClearWindowRegs\n\
+        movs r0, 0x1\n\
+        bl ClearWindowTilemap\n\
+        movs r0, 0\n\
+        bl schedule_bg_copy_tilemap_to_vram\n\
+        bl TeachyTvRemoveScrollIndicatorArrowPair\n\
+        strh r4, [r6, 0x6]\n\
+        strh r4, [r6, 0x4]\n\
+        ldr r1, _0815B384 @ =gTasks\n\
+        lsls r0, r7, 2\n\
+        adds r0, r7\n\
+        lsls r0, 3\n\
+        adds r0, r1\n\
+        ldr r1, _0815B388 @ =TeachyTvRenderMsgAndSwitchClusterFuncs\n\
+        str r1, [r0]\n\
+    _0815B37A:\n\
+        pop {r4-r7}\n\
+        pop {r0}\n\
+        bx r0\n\
+        .align 2, 0\n\
+    _0815B380: .4byte gTeachyTV_StaticResources\n\
+    _0815B384: .4byte gTasks\n\
+    _0815B388: .4byte TeachyTvRenderMsgAndSwitchClusterFuncs\n\
+        ");
+}
+#endif
+
+void TeachyTvClusFuncTransitionRenderBg2TeachyTvGraphicInitNpcPos(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    TeachyTvBg2AnimController();
+    if ( ++data[2] > 63 )
+    {
+        CopyToBgTilemapBufferRect_ChangePalette(2, gUnknown_203F450->buffer4, 0, 0, 0x20, 0x20, 0x11);
+        TeachyTvSetSpriteCoordsAndSwitchFrame(((u8*)data)[2], 8, 0x38, 7);
+        schedule_bg_copy_tilemap_to_vram(2);
+        data[2] = 0;
+        ++data[3];
+        PlayNewMapMusic(BGM_FRLG_FOLLOW_ME);
+    }
+}
+
+void TeachyTvClusFuncClearBg2TeachyTvGraphic(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    if ( ++data[2] == 134 )
+    {
+        FillBgTilemapBufferRect_Palette0(2, 0, 2, 1, 0x1A, 0xC);
+        schedule_bg_copy_tilemap_to_vram(2);
+        data[2] = 0;
+        ++data[3];
+    }
+}
+
+void TeachyTvClusFuncNpcMoveAndSetupTextPrinter(u8 taskId)
+{
+    s16 *data;
+    struct Sprite * spriteAddr;
+
+    data = gTasks[taskId].data;
+    spriteAddr = &gSprites[data[1]];
+    if ( data[2] != 35 )
+        ++data[2];
+    else {
+        if ( spriteAddr->pos2.x == 0x78 )
+        {
+            StartSpriteAnim(&gSprites[data[1]], 0);
+            TeachyTvInitTextPrinter(&gUnknown_841B83D);
+            data[2] = 0;
+            ++data[3];
+        }
+        else
+            ++spriteAddr->pos2.x;
+    }   
+}
+
+void TeachyTvClusFuncIdleIfTextPrinterIsActive(u8 taskId)
+{
+    u16* data = (u16 *)gTasks[taskId].data;
+    if ( !(sub_80BF518(0) << 16) )
```

Comment:
The shift is unnecessary.  `sub_80BF518` returns a `bool16` (if it doesn't, it should).